integration/integration_test.go

```diff
@@ -44,7 +44,29 @@ var serdecontext = json.NewContext()
 // Check the shuffled votes versus the cast votes on a few nodes
 func TestIntegration(t *testing.T) {
 	t.Run("3 nodes, 3 votes", getIntegrationTest(3, 3))
-	t.Run("10 nodes, 10 votes", getIntegrationTest(10, 10))
+	t.Run("10 nodes, 100 votes", getIntegrationTest(10, 100))
+}
+
+func TestBadVote(t *testing.T) {
+	t.Run("5 nodes, 10 votes including 5 bad votes", getIntegrationTestBadVote(5, 10, 5))
+}
+
+func TestRevote(t *testing.T) {
+	t.Run("5 nodes, 10 votes ", getIntegrationTestRevote(5, 10))
+}
+
+func TestCrash(t *testing.T) {
+	t.Run("5 nodes, 5 votes, 1 fail", getIntegrationTestCrash(5, 5, 1))
+	t.Run("5 nodes, 5 votes, 2 fails", getIntegrationTestCrash(5, 5, 2))
+	t.Run("10 nodes, 5 votes, 3 fails", getIntegrationTestCrash(10, 5, 3))
+}
+
+func TestAddNodes(t *testing.T) {
+	t.Run("5 nodes, 10 votes add 2 node during the process", getIntegrationTestAddNodes(5, 10, 2))
+}
+
+func BenchmarkIntegration(b *testing.B) {
+	b.Run("10 nodes, 100 votes", getIntegrationBenchmark(10, 100))
 }
 
 func getIntegrationTest(numNodes, numVotes int) func(*testing.T) {
```

Comment:
Random idea: You could have the # of bad votes, revotes,  crash and new nodes added as optional arguments to avoid duplicating this function for each purpose (and allow for combinations)